android/app/src/main/java/com/mattermost/rnbeta/ReceiptDelivery.java

```diff
@@ -95,26 +98,40 @@ protected static void execute(String serverUrl, String postId, String token, Str
                     .post(body)
                     .build();
 
-            try {
-                Response response = client.newCall(request).execute();
-                String responseBody = response.body().string();
-                if (response.code() != 200 || !isIdLoaded) {
-                    throw new Exception(responseBody);
-                }
-                JSONObject jsonResponse = new JSONObject(responseBody);
-                Bundle bundle = new Bundle();
-                String keys[] = new String[] {"post_id", "category", "message", "team_id", "channel_id", "channel_name", "type", "sender_id", "sender_name", "version"};
-                for (int i = 0; i < keys.length; i++) {
-                    String key = keys[i];
-                    if (jsonResponse.has(key)) {
-                        bundle.putString(key, jsonResponse.getString(key));
-                    }
+            makeServerRequest(client, request, isIdLoaded, promise);
+        }
+    }
+
+    private static void makeServerRequest(OkHttpClient client, Request request, Boolean isIdLoaded, ResolvePromise promise) {
+        try {
+            Response response = client.newCall(request).execute();
+            String responseBody = response.body().string();
+            if (response.code() != 200 || !isIdLoaded) {
+                throw new Exception(responseBody);
+            }
+            JSONObject jsonResponse = new JSONObject(responseBody);
+            Bundle bundle = new Bundle();
+            String keys[] = new String[]{"post_id", "category", "message", "team_id", "channel_id", "channel_name", "type", "sender_id", "sender_name", "version"};
+            for (int i = 0; i < keys.length; i++) {
+                String key = keys[i];
+                if (jsonResponse.has(key)) {
+                    bundle.putString(key, jsonResponse.getString(key));
                 }
-                promise.resolve(bundle);
-            } catch (Exception e) {
-                Log.e("ReactNative", "Receipt delivery failed to send");
-                promise.reject("Receipt delivery failure", e.toString());
             }
+            promise.resolve(bundle);
+            reRequestCount = 0;
```

Comment:
I don't think this would work if two notifications come in. As one notification is fetching and then you  get a second one.

I think this should be pass as an argument as I mentioned before.

Reply:
Ah. Multi-fetch case... top catch.
Handled via da85149a91d254c5c1f746003173f68f977e3725